spec/tree_spec.rb

```diff
@@ -1,14 +1,97 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
   it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+    expect(described_class.is_a? Class).to be true
+  end
+end
+
+describe 'Tree' do
+  let(:tree) { Tree.new }
+
+  it "age starts at 0" do
+    expect(tree.age).to eq 0
+  end
+
+  it "#age! increases age by one" do
+    tree.age!
+    expect(tree.age).to eq 1
+
+    tree.age!
+    expect(tree.age).to eq 2
+  end
+
+  it "height starts at 1 foot" do
+    expect(tree.height).to eq 1
+  end
+
+  it "height increases by 1 foot per year" do
+    2.times { tree.age! }
+    expect(tree.height).to eq 3
+  end
+
+  it "no apples at the beginning" do
+    expect(tree.any_apples?).to be false
+  end
+
+  it "no apples until 3 years" do
+    2.times { tree.age! }
+    expect(tree.any_apples?).to be false
+    tree.age!
+    expect(tree.any_apples?).to be true
   end
+
+  it "starts as alive" do
+    expect(tree.alive).to be true
+    expect(tree.dead?).to be false
+  end
+
+  it "tree lives 30 years" do
+    30.times { tree.age! }
+    expect(tree.dead?).to be false
+  end
+
+  it "tree does not live 31 years" do
+    31.times { tree.age! }
+    expect(tree.dead?).to be true
+  end
+
+  it "pick_an_apple! throws exception if no apples" do
+    begin
+      tree.pick_an_apple!
```

Comment:
`expect { tree.pick_an_apple! }.to raise_error NoAppleError` is a commonly used matcher.
https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher